drivers/lora/Kconfig.sx12xx

```diff
@@ -12,7 +12,6 @@ DT_COMPAT_SEMTECH_SX1262 := semtech,sx1262
 menuconfig LORA_SX12XX
 	bool "Semtech SX-series driver"
 	select HAS_SEMTECH_RADIO_DRIVERS
-	select HAS_SEMTECH_LORAMAC
```

Comment:
Splitting the Kconfig changes unrelated to the update into a separate commit would be nice.

Reply:
I thought so, but these Kconfig changes must be synchronized with CMakeLists.txt changes in loramac-node module. Overall they are required to avoid build errors.

Reply:
I just don't like unbuildable commits.

Reply:
Thats fine, it wasn't obvious that these changes aren't just cosmetic, and I'm not going through a 375 commit diff to find that out.